kwok/kwok_script_runner.go

```diff
@@ -0,0 +1,86 @@
+package kwok
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+)
+
+// GetKubeconfig get the full path to the kwok directory's
+// kubeconfig file, no matter where the functions is called from.
+func getKubeconfig() (string, error) {
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		return "", fmt.Errorf("failed to find the path of the current script")
+	}
+	dirPath := filepath.Dir(filename)
+	kubeconfigPath := filepath.Join(dirPath, "kubeconfig")
+	return kubeconfigPath, nil
+}
+
+// setKubeconfig set the kubeconfig using an environment variable.
+// It eithers get the full path to the kubeconfig, or set it to the
+// one in the kwok directory.
+func setKubeconfig(kubeconfigPath string) error {
+	if kubeconfigPath == "" {
+		kubeconfigPath, _ = getKubeconfig()
+	}
+	os.Setenv("KUBECONFIG", kubeconfigPath)
+
+	return nil
+}
+
+type ScriptExecutor interface {
+	Execute(scriptPath string) ([]byte, error)
+}
+
+type ShellScriptExecutor struct{}
+
+func (s *ShellScriptExecutor) Execute(scriptPath string) ([]byte, error) {
+	return exec.Command(scriptPath).CombinedOutput()
```

Comment:
Please avoid using CombinedOutput. Mixing STDOUT and STDERR is not a good idea and I though we've been over this enough times already.

Also I don't think you have a good reason to build an extra abstraction over `exec` here.

Reply:
This function is not new, it's just part of a file that had a namechange (I mentioned it in the PR description)... changing it is out of this PR scope.

Reply:
Its a bad excuse for keeping fragile code around.